CMakeLists.txt

```diff
@@ -41,6 +41,8 @@ option(gw_profiling "Compile a binary for profiling with NVTX markers." OFF)
 option(gw_enable_caching_allocator "Enable caching allocator." ON)
 option(gw_generate_docs "Generate Doxygen documentation" ON)
 option(gw_cuda_gen_all_arch "ON: Generate optimized CUDA code for all architectures | OFF: for detected architectures only" OFF)
+# Optionally build htslib for SAM/BAM support. Requires autoconf to be installed
+option(gw_build_htslib "Build 3rdparty htslib that allows output in SAM/BAM format" OFF)
```

Comment:
Do we make sure CI runs with it on?

Reply:
(See https://github.com/clara-parabricks/GenomeWorks/pull/556#issuecomment-703031099)
Unfortunately, CI doesn't have the required build dependencies for htslib, so it always fails during configuration :(. Is it possible for these apt dependencies be added to CI?

Reply:
@nvvishanthi I created a PR(https://github.com/clara-parabricks/gpuci-build-environment/pull/7) for adding `autoconf` to our images through Anaconda. Although, I don't think this is really needed. I was able to build & install `htslib` from a git clone by following the Basic Installation steps. https://github.com/samtools/htslib/blob/develop/INSTALL#L63-L68

Reply:
@o8191 what did you have to change to get it to build? I couldn't get it to work in CI, with 1) `./configure <args to disable some extra compression libs>` 2) `make` 3) `make install`. It said that the configure script was missing - which is what I expected since there is no autoconf.

This is from the build instructions (about the `autoheader`/`autoconf` usage):
> This step is only needed if configure.ac has been changed, or if configure
does not exist (for example, when building from a git clone).

Doesn't this apply since we're building after a git clone?

Reply:
@mimaric MR now passes in CI with it on